tests/integration/ceph_bucket_notification_test.go

```diff
@@ -122,38 +110,165 @@ func (s *ObjectSuite) TestBucketNotificationsInOrder() {
+		assert.Nil(t, err)
+		created := utils.Retry(12, 2*time.Second, "topic is created", func() bool {
+			return helper.TopicClient.CheckTopic(newTopicName)
+		})
+		assert.True(t, created)
+		err = helper.NotificationClient.CreateNotification(newNotificationName, newTopicName)
+		assert.Nil(t, err)
+		created = utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+			return helper.NotificationClient.CheckNotification(newNotificationName)
+		})
+		assert.True(t, created)
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
+		assert.Nil(t, err)
+		obc.Labels[bucketNotificationLabelPrefix+newNotificationName] = newNotificationName
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether bucket notification added
+		notificationPresent := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
+			return helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, newNotificationName, s3endpoint)
+		})
+		assert.True(t, notificationPresent)
+		err = helper.NotificationClient.DeleteNotification(newNotificationName, topicName)
+		assert.Nil(t, err)
+		err = helper.TopicClient.DeleteTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+	})
+
+	t.Run("reverse order of creating notification,topic and adding it to ObjectBucketClaim", func(t *testing.T) {
+		reverseNotificationName := "reverse-notification"
+		reverseTopicName := "reverse-topic"
+		reverseOBCName := "reverse-obc"
+		reverseBucketName := "reverse-bucket"
+		err := helper.BucketClient.CreateObcNotification(reverseOBCName, bucketStorageClassName, reverseBucketName, reverseNotificationName, true)
+		assert.Nil(t, err)
+
+		created := utils.Retry(12, 2*time.Second, "OBC is created", func() bool {
+			return helper.BucketClient.CheckOBC(reverseOBCName, "bound")
+		})
+		assert.True(t, created)
+		logger.Info("OBC created successfully")
+
+		var bkt rgw.ObjectBucket
+		i := 0
+		for i = 0; i < 4; i++ {
+			b, code, err := rgw.GetBucket(rgwcontext, reverseBucketName)
+			if b != nil && err == nil {
+				bkt = *b
+				break
 			}
-			return false
+			logger.Warningf("cannot get bucket %q, retrying... bucket: %v. code: %d, err: %v", reverseBucketName, b, code, err)
+			logger.Infof("(%d) check bucket exists, sleeping for 5 seconds ...", i)
+			time.Sleep(5 * time.Second)
+		}
+		assert.NotEqual(t, 4, i)
+		assert.Equal(t, reverseBucketName, bkt.Name)
+		err = helper.NotificationClient.CreateNotification(reverseNotificationName, reverseTopicName)
+		assert.Nil(t, err)
+		created = utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+			return helper.NotificationClient.CheckNotification(reverseNotificationName)
 		})
```

Comment:
This is verifying the notification is created, but does it expect the `CephBucketNotification` resource to process and report a status? Is this checking the Ceph RGW's notification, or the `CephBucketNotification` resource? 

I think my confusion here is a case of not having very good naming. What can we rename `CheckNotification` to in order to show it's purpose more clearly? What other methods can we create to reuse in these tests that can help others read them more easily without having to understand the deep inner workings?

Reply:
Not really, the backend notification will be created both obc and topics are ready. I will rename the `CheckNotification` into `CheckNotificationCR` and `CephBucketNotification` into `CephBucketNotificationSetonRGW`